Plugins/CoreAPI/server/graphql/resolvers/mutations.js

```diff
@@ -0,0 +1,435 @@
+import PenPal from "meteor/penpal";
+import { Mongo } from "meteor/mongo";
+import _ from "lodash";
+
+export default {
+  async setCoreAPIConfiguration(root, args, context) {
+    let currConfig = PenPal.DataStore.fetch("CoreAPI", "Configuration", {});
+    if (currConfig.length > 0) {
+      PenPal.DataStore.update(
+        "CoreAPI",
+        "Configuration",
+        { _id: `${currConfig[0]._id}` },
+        { $set: { hookURL: args.hookURL } }
+      );
+      return {
+        status: "Updated Configuration",
+        was_success: true,
+        affected_records: [currConfig[0]._id],
+      };
+    } else {
+      let addedConfig = PenPal.DataStore.insert("CoreAPI", "Configuration", {
+        hookURL: args.hookURL,
+      });
+      if (addedConfig) {
+        return {
+          status: "Inserted Configuration",
+          was_success: true,
+          affected_records: [addedConfig],
+        };
+      } else {
+        return {
+          status: "Configuration Update Failed",
+          was_success: false,
+          affected_records: [],
+        };
+      }
+    }
+  },
+  async upsertHosts(root, args, context) {
```

Comment:
I think it would make more sense to separate some of this functionality into separate files based on data type it touches. Like `configuration.mutations.js` and `hosts.mutations.js`. This will keep files from getting monolithically large

Reply:
to clarify, this is not currently a request for action